src/Common/HostResolvePool.cpp

```diff
@@ -237,7 +244,7 @@ void HostResolver::updateImpl(Poco::Timestamp now, std::vector<Poco::Net::IPAddr
     }
 
     for (auto & rec : merged)
-        if (rec.failed && rec.fail_time < last_effective_resolve)
+        if (rec.failed && rec.fail_time < now - Poco::Timespan(history.totalSeconds() * (1ull << (rec.fail_count - 1)), 0))
```

Comment:
What what expression means? `now - Poco::Timespan(history.totalSeconds() * (1ull << (rec.fail_count - 1))`
Make a variable or function, the name will give us a hint what it is.

Reply:
Moved in separate method HostResolver::Record::cleanTimeoutedFailedFlag